reducers/views/channel.js

```diff
@@ -115,11 +115,58 @@ function keepChannelIdAsUnread(state = null, action) {
     }
 }
 
+function channelPostsStatus(state = {}, action) {
+    switch (action.type) {
+    case ActionTypes.CHANNEL_POSTS_STATUS: {
+        const channelId = action.data.channelId;
+        if ((action.data).hasOwnProperty('atEnd')) {
+            return {
+                ...state,
+                [channelId]: {
+                    ...state[channelId],
+                    atEnd: action.data.atEnd,
+                },
+            };
+        }
+        return {
+            ...state,
+            [channelId]: {
+                ...state[channelId],
+                atStart: action.data.atStart,
+            },
+        };
+    }
+    default:
+        return state;
+    }
+}
+
+function channelSyncStatus(state = {}, action) {
+    switch (action.type) {
+    case GeneralTypes.WEBSOCKET_FAILURE || GeneralTypes.CLOSED: {
```

Comment:
Also, I don't think `GeneralTypes.CLOSED` exists

Reply:
This should be `GeneralTypes.WEBSOCKET_CLOSED`. Nice catch